src/components/cards/TierCard/TierCard.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+import { TIER_DATA } from '../../../../types/constants';
+import { TierCardProps } from '../../../types';
+import { CircleCheck } from '../../icons';
+import './styles.css';
+
+function TierCard({ tier, setTier, selectedTier }: TierCardProps) {
+  const [thisTier, setThisTier] = useState(new Number(tier).valueOf());
```

Comment:
why can't we just use `tier`? it's set by the parent, and no changes internally

Reply:
cant access it while tier is a string: because its accessing it via object.keys its returning it as a string. This pattern takes it wether is a string or a number an creates a number from it. I could probably move the pattern to the .map function in upgrade tier instead... thoughts?

Reply:
if we limit the `tier` to only be a number on the type, then make sure we cast it before hand - does that solve this problem? regardless, we should force the parent to cast it to a number and not allow anything else